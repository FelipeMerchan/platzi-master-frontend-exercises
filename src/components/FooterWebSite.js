import React, { Component } from 'react'

import './styles/FooterWebSite.css'
import twitterLogo from '../images/twitter.svg'
import facebookLogo from '../images/facebook.svg'
import spotifyLogo from '../images/spotify.svg'
import youtubeLogo from '../images/youtube.svg'

class FooterWebSite extends Component {
  render() {
    return (
      <footer className="FooterWebSite">
          <div className="FooterWebSite__social">
            <a href="/" className="twitter" target="_blank">
              <img width="40" src={twitterLogo} alt="Twitter" />
            </a>
            <a href="/" className="facebook" target="_blank">
              <img width="40" src={facebookLogo} alt="Facebook" />
            </a>
            <a href="/" className="spotify" target="_blank">
              <img width="40" src={spotifyLogo} alt="Spotify" />
            </a>
            <a href="/" className="youtube" target="_blank">
              <img width="40" src={youtubeLogo} alt="YouTube" />
            </a>
          </div>
          <span className="FooterWebSite__copyright">
            &copy;2012. Todos los derechos reservados
          </span>
      </footer>
    )
  }
}

export default FooterWebSite