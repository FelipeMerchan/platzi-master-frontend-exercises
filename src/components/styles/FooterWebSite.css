.FooterWebSite {
  display: flex;
  justify-content: space-between;
  align-items: center;
  height: 80px;
  font-family: var(--font-2);
  color: var(--typographyColor);
}

.FooterWebSite__social a {
  display: inline-block;
  width: 40px;
  height: 40px;
  margin-right: var(--space);
  line-height: 40px;
  font-size: 1.4em;
  text-align: center;
  border-radius: 50%;
}

.FooterWebSite__social a:last-child {
  margin-right: 0;
}

@media screen and (max-width: 768px) {
  .FooterWebSite {
    display: block;
    height: auto;
    padding: var(--space-2);
    text-align: center;
  }

  .FooterWebSite__social {
    margin-bottom: var(--space);
  }
}