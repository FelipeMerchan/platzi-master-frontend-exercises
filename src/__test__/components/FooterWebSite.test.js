import React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import '../../setupTest'
import FooterWebSite from '../../components/FooterWebSite'

describe('Probar el componente <FooterWebSite />', () => {
  const footerWebSite = shallow(<FooterWebSite />)

  test('Render del componente FooterWebSite', () => {
    expect(footerWebSite.length).toEqual(1)
  })
  test('Render de los las RRSS', () => {
    expect(footerWebSite.find('.FooterWebSite__social').children()).toHaveLength(4);
  });
  test('Render de los derechos de autor', () => {
    expect(footerWebSite.find('.FooterWebSite__copyright').text()).toEqual('Â©2012. Todos los derechos reservados')
  })
  test('Render con el className facebook', () => {
    expect(footerWebSite.exists('.facebook')).toBeTruthy()
  })
})

describe('FooterWebSite Snapshot', () => {
  test('Comprobar la UI del componente FooterWebSite', () => {
    const footerWebSite = create(<FooterWebSite />)
    expect(footerWebSite.toJSON()).toMatchSnapshot()
  })
})