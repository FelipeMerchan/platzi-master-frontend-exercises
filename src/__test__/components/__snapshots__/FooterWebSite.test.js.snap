// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterWebSite Snapshot Comprobar la UI del componente FooterWebSite 1`] = `
<footer
  className="FooterWebSite"
>
  <div
    className="FooterWebSite__social"
  >
    <a
      className="twitter"
      href="/"
      target="_blank"
    >
      <img
        alt="Twitter"
        src="twitter.svg"
        width="40"
      />
    </a>
    <a
      className="facebook"
      href="/"
      target="_blank"
    >
      <img
        alt="Facebook"
        src="facebook.svg"
        width="40"
      />
    </a>
    <a
      className="spotify"
      href="/"
      target="_blank"
    >
      <img
        alt="Spotify"
        src="spotify.svg"
        width="40"
      />
    </a>
    <a
      className="youtube"
      href="/"
      target="_blank"
    >
      <img
        alt="YouTube"
        src="youtube.svg"
        width="40"
      />
    </a>
  </div>
  <span
    className="FooterWebSite__copyright"
  >
    Â©2012. Todos los derechos reservados
  </span>
</footer>
`;
